datafusion/optimizer/src/optimize_projections.rs

```diff
@@ -1192,4 +1354,126 @@ mod tests {
         \n    TableScan: test projection=[a]";
         assert_optimized_plan_equal(&plan, expected)
     }
+
+    // Optimize Projections Rule, pushes down projection through users defined logical plan node.
+    #[test]
+    fn test_user_defined_logical_plan_node() -> Result<()> {
```

Comment:
As written it is hard to quickly understand what cases each test is covering.

Would it be possible to add a comment to the tests explaining what they are testing (or maybe update the name)?

Reply:
Sure. Added comment explaining the rationale.